src/FixedDataTableCellGroup.js

```diff
@@ -76,25 +87,70 @@ var FixedDataTableCellGroupImpl = React.createClass({
     }, false);
 
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
+    var count = 0;
+
+    var newColumnsToRender = new Array(props.maxVisibleColumns);
+    var positions = new Array(props.maxVisibleColumns);
+
+    for (var i = 0; i < columns.length; i++) {
       var columnProps = columns[i].props;
       var recycable = columnProps.allowCellsRecycling && !isColumnReordering;
       if (!recycable || (
-            currentPosition - props.left <= props.width &&
-            currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
+          currentPosition - props.left <= props.width &&
+          currentPosition - props.left + columnProps.width >= 0)) {
+        positions[i] = currentPosition;
+        newColumnsToRender[count++] = i;
       }
+
       currentPosition += columnProps.width;
     }
+
+    //TODO move this recycle logic into main state
+    const newColumnsSet = new Set(newColumnsToRender);
+    const oldColumnsSet = new Set(this._columnsToRender);
+    const indexes = [];
+
+    for (var i = props.maxVisibleColumns; i >= 0; i--) {
+      var column = this._columnsToRender[i];
+      if (!column || !newColumnsSet.has(column)) {
+        indexes.push(i);
+      }
+    }
+
+    newColumnsToRender.forEach((column) => {
+      if (!oldColumnsSet.has(column)) {
+        this._columnsToRender[indexes.pop()] = column;
```

Comment:
Not sure if there is a better way to do this, but we need to find indexes that we can change.

last render: [1, 2, 3, 4, 5]
new columns to render: [2, 3, 4, 5, 6]
Optimally, we want to render [6, 2, 3, 4, 5] To no remount all the columns.

To do this, we create sets of old columns and new columns.
Then we loop through our last columns, and store indexes where we don't need the column.
For example, we see column 1 in our last render is not in the new column set, so we store index 0 in indexes

Then we loop through our new columns. If we find a column not in our old column set, we take one of the indexes and store that column there.